routers/api/v1/repo/issue.go

```diff
@@ -55,12 +61,36 @@ func ListIssues(ctx *context.APIContext) {
 		isClosed = util.OptionalBoolFalse
 	}
 
-	issues, err := models.Issues(&models.IssuesOptions{
-		RepoIDs:  []int64{ctx.Repo.Repository.ID},
-		Page:     ctx.QueryInt("page"),
-		PageSize: setting.UI.IssuePagingNum,
-		IsClosed: isClosed,
-	})
+	// Define issues var
+	var issues []*models.Issue
+
+	// Check for search
+	keyword := strings.Trim(ctx.Query("q"), " ")
+	if bytes.Contains([]byte(keyword), []byte{0x00}) {
+		keyword = ""
+	}
+	var issueIDs []int64
+	var err error
+	if len(keyword) > 0 {
+		issueIDs, err = indexer.SearchIssuesByKeyword(ctx.Repo.Repository.ID, keyword)
+
+		// Didn't found anything
+		if len(issueIDs) == 0 {
```

Comment:
Not needed, `issues` is already empty

Reply:
But if I don't define it there, I cannot access it in line 83 and 89

Reply:
It is defined on line 65

Reply:
I am talking about variable `issues` and `if` block.

Reply:
Ups, I though you were talking about defining the `issues` variable in line 65 :grin: